frontend/public/src/lib/courseApi.js

```diff
@@ -52,3 +52,9 @@ export const generateStartDateText = (run: CourseRunDetail) => {
 export const isFinancialAssistanceAvailable = (run: CourseRunDetail) => {
   return run.page ? !!run.page.financial_assistance_form_url : false
 }
+
+export const isUpgradable = (run: CourseRunDetail) => {
```

Comment:
Since you already have this ;logic on the backend (`CourseRun.is_upgradable`), we should just serialize that in addition to the deadline itself and use that on the frontend. That way if the logic ever changes there's only one place to do so.

Reply:
This makes sense. I've applied this while addressing feedback.